src/hooks/useAllCities.js

```diff
@@ -1,28 +1,29 @@
-import { useMemo } from 'react';
+import { useEffect, useMemo } from 'react';
 import { useInfiniteQuery } from '@tanstack/react-query';
 import fetchCities from '../services/CitiesService';
 import transformCityItem from '../data/transformCityItem';
 
 const ALL_CITIES_QUERY_KEY = 'allCities';
 
-const useAllCities = ({ pageSize, sortOrder }) => {
+const useAllCities = ({ pageSize, sortOrder, searchQuery }) => {
   const {
     data,
     error,
     isLoading,
     fetchNextPage,
     hasNextPage,
     isFetchingNextPage,
+    refetch,
   } = useInfiniteQuery({
-    queryKey: [ALL_CITIES_QUERY_KEY],
+    queryKey: [ALL_CITIES_QUERY_KEY, searchQuery],
```

Comment:
Sooo, I've investigated this.

`searchQuery` in `queryKey` + `staleTime` will work nicely without manually refetch

Reply:
Changing key will lead to refetch automatically and leaving `staleTime` will help to save cached data without sending extra queries to the BE

Reply:
![image](https://github.com/user-attachments/assets/95f86fd2-42f2-46d5-bc39-ca39b02da3d5)